pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -572,6 +572,174 @@ var _ = Describe("Snapshotter", func() {
+		})
+
+		Context("Previous full snapshot was taken exactly at scheduled snapshot time, no FullSnapshot was missed", func() {
+			It("should return false", func() {
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 1 days before at exactly at scheduled time
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-1, (currentHour+2)%24, (currentMin+1)%60, 0, 0, time.Local),
+				}
+
+				isFullSnapMissed := ssr.IsTakingFullSnapRequiresAtStartup(fullSnapshotTimeWindow)
+				Expect(isFullSnapMissed).Should(BeFalse())
+			})
+		})
+
+		Context("Previous snapshot was taken within 24hrs and next schedule full-snapshot will be taken within 24hs of time window", func() {
+			It("should return false", func() {
+				scheduleHour := (currentHour + 4) % 24
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", currentMin, scheduleHour),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 4hrs 10 mins before startup of backup-restore
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), (currentHour-4)%24, (currentMin-10)%60, 0, 0, time.Local),
+				}
+				isFullSnapCanBeMissed := ssr.IsTakingFullSnapRequiresAtStartup(fullSnapshotTimeWindow)
+				Expect(isFullSnapCanBeMissed).Should(BeFalse())
+			})
+		})
+
+		Context("Previous snapshot was taken within 24hrs and next schedule full-snapshot likely to cross 24hs of time window", func() {
+			It("should return true", func() {
+				scheduleHour := (currentHour + 8) % 24
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", currentMin, scheduleHour),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 18hrs(<24hrs) before startup of backup-restore
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), (currentHour-18)%24, (currentMin)%60, 0, 0, time.Local),
+				}
+				isFullSnapCanBeMissed := ssr.IsTakingFullSnapRequiresAtStartup(fullSnapshotTimeWindow)
+				Expect(isFullSnapCanBeMissed).Should(BeTrue())
+			})
+		})
+	})
+
+	FDescribe("Scenarios to get maximum time window for full snapshot", func() {
```

Comment:
```suggestion
	Describe("Scenarios to get maximum time window for full snapshot", func() {
```

Reply:
thanks for pointing this out. I forgot to remove this